src/py/flwr/server/strategy/aggregate.py

```diff
@@ -91,6 +91,90 @@ def aggregate_krum(
     return weights[np.argmin(scores)]
 
 
+BYZANTINE_RESILIENT_SINGLE_RET_MODEL_AGGREGATION = [
+    aggregate_krum
+]  # also GeoMed (but not implemented yet)
+BYZANTINE_RESILIENT_MANY_RETURN_MODELS_AGGREGATION = (
+    []
+)  # Brute, Medoid (but not implemented yet)
+
+
+# pylint: disable=too-many-locals
+def aggregate_bulyan(
+    results: List[Tuple[NDArrays, int]],
+    num_malicious: int,
+    aggregation_rule: Callable,
```

Comment:
@danieljanes Could you give me some feedback regarding the proposed solution, mainly regarding the aggregation_rule parameter? 

The Bulyan uses the Byzantine resilient aggregation internally, and the specific aggregation choice should be up to the user. This functionality comes with complex hint typing for callable that might not cover all choices of aggregation rule (introduced in the future). I'm not sure if I should follow by trying to capture the possibilities or maybe you have any suggestions for a neater solution for this problem. I am not satisfied with the list in the middle of the file that captures the different types of Byzantine resilient aggregation (with the division implied by the Bulyan aggregation method).

Reply:
@adam-narozniak I agree that enumerating all the possible typing options would be too complex, let's stick to just `Callable`.

Regarding the two list constants, since they are used in only one place, should we just inline the possible values there?